rain_api_core/egress_util.py

```diff
@@ -150,32 +144,36 @@ def process_request(varargs, b_map):
     return path, bucket, object_name, headers
 
 
-def check_private_bucket(bucket, b_map):
-
+def check_private_bucket(bucket, b_map, optional_uri=""):
     log.debug('check_private_buckets(): bucket: {}'.format(bucket))
 
     # Check public bucket file:
     if 'PRIVATE_BUCKETS' in b_map:
         for priv_bucket in b_map['PRIVATE_BUCKETS']:
-            if bucket == prepend_bucketname(priv_bucket):
+            if bucket_prefix_ismatch(bucket, priv_bucket, optional_uri):
                 # This bucket is PRIVATE, return group!
                 return b_map['PRIVATE_BUCKETS'][priv_bucket]
 
     return False
 
 
-def check_public_bucket(bucket, b_map):
-
+def check_public_bucket(bucket, b_map, optional_uri=""):
     # Check for PUBLIC_BUCKETS in bucket map file
     if 'PUBLIC_BUCKETS' in b_map:
         log.debug('we have a PUBLIC_BUCKETS in the ordinary bucketmap file')
         for pub_bucket in b_map['PUBLIC_BUCKETS']:
-            #log.debug('is {} the same as {}?'.format(bucket, prepend_bucketname(pub_bucket)))
-            if bucket == prepend_bucketname(pub_bucket):
+            # log.debug('is {} the same as {}?'.format(bucket, prepend_bucketname(pub_bucket)))
+            if bucket_prefix_ismatch(bucket, pub_bucket, optional_uri):
```

Comment:
It looks like you dropped the call to prepend_bucketname(). What happens when buckets have a bucket map prefix now?

Reply:
`Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> bucket_check='bucket1'
>>> bucket_map='bucket1/prefix1/'
>>> object='prefix1/path/object'
>>> if bucket_check == bucket_map.split('/')[0] and object.startswith("/".join(bucket_map.split('/')[1:])):
...     print(True)
... 
True
>>> bucket_check='bucket2'
>>> bucket_map='bucket1/prefix1'
>>> object='prefix1/path/object'
>>> if bucket_check == bucket_map.split('/')[0] and object.startswith("/".join(bucket_map.split('/')[1:])):
...     print(True)
... 
>>> 
`

Reply:
I copied and pasted the test. I got back true as expected for a match and returned nothing as expected for a mismatch with bucket maps with a prefix

Reply:
Object prefixes, what we’re trying to support, are different from Bucket name prefixes, which is what `prepend_bucketname()` accounted for.